src/amo/components/Home/index.js

```diff
@@ -144,8 +144,10 @@ export class HomeBase extends React.Component {
               to={`/collections/mozilla/${collectionSlug}/`}
               className="Home-SubjectShelf-link"
             >
-              <Icon name={`Home-SubjectShelf-${collectionSlug}`} />
-              <span>{title}</span>
+              <Icon name={`Home-SubjectShelf-${collectionSlug}`}>
```

Comment:
hi @hritvi : ) I *think* we can maybe keep the same markup and still style this using flex / grid. Have you tried that? It's ok to change markup when needed but I am just wondering if this is really needed here..? 

Also, just curious - what does "vw" stand for in `Home-SubjectShelf-link-text-vw`?

Reply:
Ya, I tried that but could not get a satisfactory result @rebmullin. Any advice by you would be of great help.

'vw' stands for viewportwidth, the content in <code>Home-SubjectShelf-link-text-vw</code>will be displayed according to the viewport width, that's why the name. Here too, any other name suggested by you would help. Thanks

Reply:
does something like this work?

html:
```html
...
 <Icon name={`Home-SubjectShelf-${collectionSlug}`} />
 <span className="Home-SubjectShelf-text">{title}</span>
...
```

```css
.Home-SubjectShelf-list-item {
  flex-grow: 1;
  list-style-type: none;
  margin: 15px 0;
  text-align: center;
  width: 100%;

  @include respond-to(medium) {
    width: 40%;
  }

  @include respond-to(large) {
    margin: 0;
  }
}

.Home-SubjectShelf-link {
  font-size: $font-size-m-smaller;
  text-decoration: none;
  display: flex;
  align-items: center;

  @include respond-to(large) {
    display: inline-block;
  }

  &:link,
  &:visited {
    color: $black;
  }

  &:active,
  &:focus,
  &:hover {
    color: $link-color;
    text-decoration: underline;
  }
}

.Home-SubjectShelf-list-item .Icon {
  background-position: left;
  background-repeat: no-repeat;
  background-size: 64px;
  border-radius: $border-radius-default;
  content: '';
  display: block;
  flex-grow: 1;
  height: 64px;
  margin-bottom: 5px;
  transform: scale(0.8);

  @include respond-to(medium) {
    background-position: center;
    transform: initial;
    width: auto;
  }

  @include respond-to(large) {
    width: 100%;
  }
}

.Home-SubjectShelf-text {
  // 70px accounts for approx width of Icon plus some cushion
  width: calc(100% - 70px);
  text-align: left;

  @include respond-to(medium) {
    // 100px accounts for approx width of Icon plus some cushion
    width: calc(100% - 100px);
  }

  @include respond-to(large) {
    width: auto;
  }
}
```
sorry I am not showing diff here but something like seems to work. Basically I am toggling widths and display at different breakpoints.

You might have to adjust it more to get it to match the comps better.

Reply:
Thanks for your help :smiley: 
I tried that but it doesn't seem to work as it should be.
![screenshot from 2018-09-06 11-01-26](https://user-images.githubusercontent.com/32164618/45136963-44c0ea00-b1c4-11e8-911c-e1402aed1bb1.png)

I agree with you on toggling widths but we also have to toggle height where the text would lie and to make it lie in the center, I suppose, we do have to add some extra HTML.

Reply:
hmmm.. this is what I see on my end:
![Alt text](https://monosnap.com/image/POizWgVwnhHcs590o5W8PX21dPM8o0.png)

can you see that you have flex and align-items styles added to the link class (with the html I have noted above):

```css
.Home-SubjectShelf-link {
...
  display: flex;
  align-items: center;
...
```

Reply:
It worked @rebmullin with some minor changes. Thanks :smile: